AFQ/api.py

```diff
@@ -392,12 +451,26 @@ def log_and_save_trk(self, sft, fname):
         self.logger.info(f"Saving {fname}")
         save_tractogram(sft, fname, bbox_valid_check=False)
 
+    def _get_data_gtab(self, row, filter_b=True):
+        img = nib.load(row['dwi_file'])
+        data = img.get_fdata()
+        gtab = row['gtab']
+        if filter_b and (self.min_bval is not None):
+            data = data[
+                ...,
+                (gtab.bvals >= self.min_bval)
+                or (gtab.bvals <= self.b0_threshold)]
+        if filter_b and (self.max_bval is not None):
+            data = data[
+                ...,
+                (gtab.bvals <= self.max_bval)
+                or (gtab.bvals <= self.b0_threshold)]
+        return data, gtab, img
```

Comment:
Sorry: one more thing -- shouldn't we also create a new GradientTable object that has only the relevant b-values and corresponding b-vectors?

Reply:
Good catch. I just rewrote this function in a way which I think makes more sense.